v3/pb.go

```diff
@@ -513,8 +520,9 @@ func (pb *ProgressBar) Err() error {
 
 // String return currrent string representation of ProgressBar
 func (pb *ProgressBar) String() string {
-	res, _ := pb.render()
-	return res
+	res, width := pb.render()
+	repeatCnt := pb.Width() - width
```

Comment:
What reason to do additional align here? Extra call Width() makes a syscall and this is a relatively not cheap operation. I think we need to calculate width once per "Pool.print"
Also this breaks `TestPBTemplate`

Reply:
I've tried to create a test case to show the issue, but failed. Here's what I was doing:
As per the pool example, I was using a template with just a text string (strange I know to not have a progress bar but it was a useful way to format the output by using a text only entry as a status). To be clear, literally just {{msg}} as the template, and then populate that.
I was writing a long string to the msg, then a shorter one. The print printed the length of the message, so old characters did not get overwritten. This meant the display was corrupted with stale data.
I think normal progress bars do not see this as they are sized to fill the full width as part of the progress bar.

I'm happy to accept the argument I'm abusing the interface, so I should format the message to be full width, but that then made the interface tricky for me, as the source of the message did not have that information easyily available, but here did.

I considered a new implementation of pb that was effectively an extention of ProgressBar that would be text message only that would then encapsulate this and solve the issue only for the bits that needed it.
But I was being lazy. I wonder if you have a neater solution?

Reply:
I think we need to do it in `pool.print` because `bar.String` is just a string representation and does not assume some paddings (it can be used not only for terminal).
For me, the best solution will be to do all work in `pool.print`:
1. calculate width once
2. set width to each bar
3. add padding to each `bar.String` result if needed (maybe via some helper in a pool.go file)

Reply:
Okay I have this on my branch, but it causes an issue in testing as the expected test data for those examples is hard to generate. I can either kill those tests and move them to the readme as you suggest, or provide a mechanism to overrid the nbar width read from the terminal width.

Reply:
Resolving as discussing in other thread.